### PR TITLE
Set index as default x axis for DataFrame

### PR DESCRIPTION
**Pull request by matugm**
Hi,
this is my attempt at fixing https://github.com/SciRuby/daru/issues/402.

It breaks one test, but otherwise it seems to work.

If you have a better idea on how to do this let me know :)

### CODE REVIEW

File: lib/daru/plotting/nyaplot/dataframe.rb [hunk 1]
```diff
@@ -21,6 +21,11 @@ module NyaplotLibrary
         #   df = Daru::DataFrame.new({a:['A', 'B', 'C', 'D', 'E'], b:[10,20,30,40,50]})
         #   df.plot type: :bar, x: :a, y: :b
         def plot opts={}, &block
+          unless opts[:x]
+            opts[:x] = :_index
+            self.[]=(:_index, @index.to_a)
```
Comment (zverok) [inline lib/daru/plotting/nyaplot/dataframe.rb]: Why not `self[:_index] = @index.to_a`?

**Comment by v0dro** [comment]
@matugm ping!

**Comment by matugm** [comment]
Will work on this today.

**Comment by matugm** [comment]
Ok I made the requested changes. Only thing left is 2 Rubocop warnings.

File: lib/daru/plotting/nyaplot/dataframe.rb [hunk 2]
```diff
@@ -30,6 +32,15 @@ def plot opts={}, &block
 
         private
 
+        def x_axis_undefined?(opts)
+          !(opts[:x] || opts.keys.any? { |k| k.to_s.match(/x\d+/) })
```
Comment (athityakumar) [inline lib/daru/plotting/nyaplot/dataframe.rb]: Just a minor nit-picking from my part. 😄 

This format ...
```ruby
x_axis_defined = cond_1 || cond_2
index_as_default_x_axis(opts) unless x_axis_defined
```
... seems a bit more readable, when compared to the current format ...
```ruby
x_axis_undefined = !(cond_1 || cond_2)
index_as_default_x_axis(opts) if x_axis_undefined
```

Comment (matugm) [reply lib/daru/plotting/nyaplot/dataframe.rb]: Ok, done.

File: lib/daru/plotting/nyaplot/dataframe.rb (hunk 1 above)
Comment (matugm) [reply lib/daru/plotting/nyaplot/dataframe.rb]: No specific reason. I made the change :)

**Comment by zverok** [comment]
@matugm But where's any specs for the new behavior?..

**Comment by matugm** [comment]
Spec added & also tested manually with IRuby.

**Comment by zverok** [comment]
Thanks!

**Pull request merged**